File: 001-merge-alternatively/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i = 0, j = 0;
        while(i < word1.length() && j < word2.length()) {
            if(i == j) {
                sb.append(word1.charAt(i));
                i += 1;
            } else {
                sb.append(word2.charAt(j));
                j += 1;
            }
        
        while(i < word1.length()) {
            sb.append(word1.charAt(i));
            i += 1;
        }
        while(j < word2.length()){
            sb.append(word2.charAt(j));
            j += 1;
        }
        return sb.toString();
    }
}

File: 001-merge-alternatively/Solution.py
class Solution:
    # https://leetcode.com/problems/greatest-common-divisor-of-strings/submissions/
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ""
        i = 0
        j = 0

        while i <= len(word1) - 1 and j <= len(word2) - 1:
            char1 = word1[i]
            char2 = word2[j]
            result += (char1 + char2)
            i += 1
            j += 1
        else:
            while i <= len(word1) - 1:
                result += word1[i]
                i += 1
            
            while j <= len(word2) - 1:
                result += word2[j]
                j += 1
      
        return result